enhancements/installer/credentials-management-outside-openshift-cluster.md

```diff
@@ -0,0 +1,130 @@
+level cloud credential. The admin credential is stored in kube-system namespace and then used by the cloud credential 
+operator to process the CredentialRequests in the cluster and create new users with fine-grained permissions.
+The customers have reported that the need to store admin credentials inside the cluster is a major disadvantage. Based 
+on the feedback, we propose to run credentials related setup outside a cluster and then start installation process in a 
+*manual* mode without providing admin credentials to the installer.
+
+### User Stories
+
+#### Story 1
+As an OpenShift cluster administrator I should be able to extract the CredentialsRequest manifests from the release image 
+and create required manifests/infrastructure to satisfy all in-cluster component's CredentialsRequests.
+
+#### Story 2
+As an OpenShift cluster administrator, I should be able to reconcile my credentials manually using local CLI tooling on an on-going basis.
+
+#### Story 3
+As an OpenShift cluster administrator, I should be able to prep credentials for a cluster upgrade.
+
+### Risks and Mitigations
+
+## Design Details
+
+We intend to build an optional tool **ccoctl** which will handle credentials management of the cluster in *manual* mode. 
+The following is the set of requirements for the current prototype (for AWS). We can have something similar for other 
+platforms.
+
+* ccoctl should be able to setup Identity Provider to authenticate OpenShift components
+* ccoctl should be able to take a list of CredentialsRequests from the release image and create/update 
+  Roles/Users/ServicePrincipals/ServiceAccounts with appropriate permissions.
+* ccoctl should be able to take list of CredentialsRequests for a release image, and the Identity Provider URL to generate 
+  the objects that need to be passed to the installer for successful installation
+* ccoctl should be able to delete all the resources that it had created in the cloud
+
+We envision **ccoctl** as a recommended tool to setup credentials on new platforms but customers are free to use other 
+tools like Terraform/AWS CloudFormation to do the above mentioned tasks. Read more about AWS implementation details [here](https://github.com/openshift/cloud-credential-operator/blob/master/docs/ccoctl.md)
+
+We also plan to have a detailed documentation in place to guide new cloud providers to implement *manual* mode.
+
+To upgrade a cluster we need to execute following steps:
+* Examine the CredentialsRequests in the new OpenShift release. Check if permissions in the existing CredentialsRequest
+  have changed.
+* Create/update credentials in the underlying cloud provider, and also create/update Kubernetes Secrets in the correct
+  namespaces to satisfy all CredentialsRequests in the new release.
+* Set an appropriate annotation `cloudcredential.openshift.io/upgradeable-to` to a new upgradable version.
```

Comment:
Can we please add how this requirement is enforced? CCO sets `Upgradeable=False` whenever the next minor isn't included in the value?

Reply:
CCO sets `Upgradeable=False` by default in Manual mode. When user is ready to upgrade (after making necessary changes to satisfy credentials requests in next version), they can `cloudcredential.openshift.io/upgradeable-to:<version>` to upgrade